observer/observer_test.go

```diff
@@ -56,3 +55,123 @@ func TestObserver_IsConcurrentJobsLimitReached(t *testing.T) {
 	isLimitReached = o.IsConcurrentJobsLimitReached(v1alpha1.ArchiveType, 0)
 	assert.False(t, isLimitReached)
 }
+
+func TestObserver_AreAllCallbacksInvoked(t *testing.T) {
+	for expectedInvocations := 0; expectedInvocations < 3; expectedInvocations++ {
+		t.Run(fmt.Sprintf("Given%dCallbacks", expectedInvocations), func(t *testing.T) {
+			// Given
+			var actualInvocations = new(int)
+			ojc := func(job ObservableJob) {
+				*actualInvocations++
+			}
+
+			callbacks := make([]ObservableJobCallback, 0)
+			for i := 0; i < expectedInvocations; i++ {
+				callbacks = append(callbacks, ojc)
+			}
+
+			oj := ObservableJob{
+				Job: &batchv1.Job{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "some-name",
+						Namespace: "default",
+					},
+				},
+				JobType:    v1alpha1.BackupType,
+				Event:      Suceeded,
+				Exclusive:  true,
+				Repository: "some-repo",
+				callbacks:  callbacks,
+			}
+
+			observedJobs := map[string]ObservableJob{
+				"default/some-name": oj,
+			}
+
+			o := &Observer{
+				events:       make(chan ObservableJob, 10),
+				observedJobs: observedJobs,
+				log:          ctrl.Log.WithName("observer-test"),
+				mutex:        sync.Mutex{},
+			}
+
+			// When
+			o.handleEvent(oj)
+
+			// Then
+			assert.Equal(t, expectedInvocations, *actualInvocations)
+		})
+	}
+}
+
+func TestObserver_AreOnlyExpectedCallbacksInvoked(t *testing.T) {
+	tests := map[string]struct {
+		givenEventType   EventType
+		expectInvocation bool
+	}{
+		"GivenStatusSucceeded": {
+			givenEventType:   Suceeded,
+			expectInvocation: true,
+		},
```

Comment:
I was referring to this construct:
```suggestion
		"GivenStatusSucceeded_WhenHandleEvent_ThenExpectInvocation": {
			givenEventType:   Suceeded,
			expectInvocation: true,
		},
```
It might look a bit verbose and tedious to copy-paste, but if a test fails on GH action or command line, that sometimes gives better indication which test failed and why. Often the assertion output can be confusing. Also for consistency reasons (we have other tests following this scheme where reasonablee)

Reply:
I've adjusted to these messages.